vminitd/Sources/vminitd/TerminalIO.swift

```diff
@@ -36,30 +40,47 @@ final class TerminalIO: ManagedProcess.IO & Sendable {
         stdio: HostStdio,
         log: Logger?
     ) throws {
-        let pair = try Terminal.create()
-        self.parent = pair.parent
-        self.child = pair.child
         self.stdio = stdio
         self.log = log
 
-        let ptyHandle = child.handle
-        let useHandles = stdio.stdin != nil || stdio.stdout != nil
-        // We currently set stdin to the controlling terminal always, so
-        // it must be a valid pty descriptor.
-        process.stdin = useHandles ? ptyHandle : nil
-
-        let stdoutHandle = useHandles ? ptyHandle : nil
-        process.stdout = stdoutHandle
-        process.stderr = stdoutHandle
+        process.stdin = nil
+        process.stdout = nil
+        process.stderr = nil
     }
 
     func resize(size: Terminal.Size) throws {
         if self.stdio.stdin != nil {
-            try parent.resize(size: size)
+            try parent?.resize(size: size)
         }
     }
 
-    func start() throws {
+    func start() throws {}
+
+    func attach(pid: Int32, fd: Int32) throws {
+        #if os(linux)
+        let containerFd = Glibc.syscall(Int(SYS_pidfd_open), pid, 0)
```

Comment:
We only build against musl today, the entire guest userland is just two statically linked executables (vminitd and the vmexec one you've been making most of the changes in). I'm fine supporting glibc if anyone wants to craft their own userland that's more full featured, but you'll need to do something like this here https://github.com/apple/containerization/blob/1a59de82b052a86edd9e2bd6f023d212a061b0b4/Sources/ContainerizationOCI/Bundle.swift#L20-L28.